waltz-tools/src/test/java/com/wepay/waltz/tools/storage/StorageCliTest.java

```diff
@@ -84,57 +88,86 @@ Properties createProperties(String connectString, String znodePath, SslSetup ssl
-        WaltzStorage storage = storageRunner.awaitStart();
-        int adminPort = storage.adminPort;
-        String adminConnectString = helper.getHost() + ":" + adminPort;
-
-        helper.startWaltzServer(true);
+        String[] adminConnectStrings = new String[numStorages];
+        int adminPort = 0;
+        for (int s = 0; s < numStorages; s++) {
+            WaltzStorageRunner storageRunner = helper.getWaltzStorageRunner(s);
+            storageRunner.startAsync();
+            WaltzStorage storage = storageRunner.awaitStart();
+            adminPort = storage.adminPort;
+            adminConnectStrings[s] = helper.getHost() + ":" + adminPort;
+        }
 
         // wait for storage node partitions creation
         Thread.sleep(1000);
 
-        // default partition assignment is set to True
-        PartitionInfoSnapshot info = storage
-                .getPartitionInfos()
-                .stream()
-                .filter(pis -> pis.partitionId == partitionId)
-                .findFirst()
-                .get();
-        assertFalse(info.isAssigned);
+        String expectedCmdOutput;
 
         try {
+            int index = 0;
+            int partitionsPerStorage = numPartitions / numStorages;
             // add partition to storage
-            String[] args1 = {
-                    "add-partition",
-                    "--storage", adminConnectString,
-                    "--partition", String.valueOf(partitionId),
-                    "--cli-config-path", configFilePath
-            };
-
-            StorageCli.testMain(args1);
+            for (int s = 0; s < numStorages; s++) {
+                for (int p = 0; p < partitionsPerStorage; p++) {
+                    String[] addPartitionArgs = {
+                            "add-partition",
+                            "--storage", adminConnectStrings[s],
+                            "--partition", String.valueOf(index++),
+                            "--cli-config-path", configFilePath
+                    };
+                    StorageCli.testMain(addPartitionArgs);
+                }
+            }
 
             // successful path
-            String[] args2 = {
+            for (int s = 0; s < numStorages; s++) {
+                String[] args2 = {
+                        "list",
+                        "--storage", adminConnectStrings[s],
+                        "--cli-config-path", configFilePath
+                };
+                ByteArrayOutputStream successfulPathOut = new ByteArrayOutputStream();
+                ByteArrayOutputStream successfulPathErr = new ByteArrayOutputStream();
+                setUpStreams(successfulPathOut, successfulPathErr);
+                StorageCli.testMain(args2);
+
+                expectedCmdOutput = "Partition Info for id: " + partitionId;
+                assertTrue(successfulPathOut.toString("UTF-8").contains(expectedCmdOutput));
+            }
+            setUpStreams();
+
+
+            // no path
```

Comment:
nit: 
1) extra line above
2) suggest use either `//` or `/* */`, it is a little bit weird to see them together
3) btw, what is `no path`?

Reply:
in the existing test, `success path` references a storage address that exists and can be listed and `failure path` references a storage address that doesn't exist and thus can not be listed. Following this convention, `no path` references no address specified when calling the command.

Reply:
I see, maybe we can put something more explicit, like `no storage specify`